Mongodb Context Skeleton + Mongo Context/Model Tests

townerr opened the pull request:
Moved the db context to a testing branch which contains skeleton code for the context and tests for its methods.

townerr committed:
Initial mongo collection models and settings class
Added mongo nuget packages

townerr committed:
Add Mongo startup confid
Add mongo settings to user secrets
Added notification collection to user

townerr committed:
Removed total unread notifications

townerr committed:
Removed DB settings from being a service in startup.
Added Notification Repo and Interface.
Added db settings to repo.
Change users in notificaion model to be email ids.
Moved non sensitive stuff from user secret to appsettings.

townerr committed:
Removed all of db settings service config

townerr committed:
Added a NotificationsRepo for settings and collections
Removed settings stuff from Repo and replaced witr context.

townerr committed:
Merge branch 'us-notifications' into ft-codefirstdb

townerr committed:
Mongo Repo Tests

mattmarnien committed:
updated dockerfile to take dataccessprj into account

townerr committed:
add some seed data json files for mongodb

SukafuDesu committed:
[Enhancement] Added Skeleton for FakebookNotifications (#11)

* [Enhancement] Added Skeleton for FakebookNotifications

* Added Interfaces
* Added Domain Class for User and Notifications

* [Enhancement] Updated Repos According to Docs
* https://docs.google.com/document/d/1AuPOR2f3VtaDPO6iRbpaVjzjzYFbjV3-m-VCY6U3vGA/edit
* Updated NotificationsRepo to implement interface
* Create/Updated UserRepo to implement interface

* Made Repos public

SukafuDesu committed:
[Updated Dockerfile]

* Dockerfile should now build with new domains library

townerr committed:
Changed Repo to get context instead of creating a new one.
Configured db settings to get from appsettings.
Added repo and context to be servivces for dependency injection.
Removed deprecated package.

townerr committed:
Removed domain code from branch

townerr committed:
remove domain stuff

townerr committed:
fixed docker file

townerr committed:
fix docker error

townerr committed:
Added mongo tests

Commit:
Merge branch 'us-notifications' into ft-codefirstdb

townerr committed:
Removed tests to be added to a test branch

townerr committed:
Merge branch 'ft-codefirstdb' of https://github.com/2011-fakebook-project3/notifications into ft-codefirstdb

townerr committed:
Initial mongo tests

townerr committed:
Remove unneeded repo

townerr committed:
Redo mongo db tests

townerr committed:
Changed context to be only skeleton code

townerr committed:
Removed repo from startup to fix build error since it hasnt been added yet.

townerr committed:
Added tests to cover the database models

townerr committed:
Added db settings unit tests

escalonn reviewed (changes requested):
the branch is not up to date, fix that

townerr committed:
merge

townerr committed:
fix sln

townerr committed:
Fix moq reference

escalonn commented on FakebookNotifications/FakebookNotifications.Testing/FakebookNotifications.Testing.csproj:
```diff
@@ -23,6 +22,7 @@
 
 	<ItemGroup>
 		<ProjectReference Include="../FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj" />
+    <ProjectReference Include="..\FakebookNotifications.DataAccess\FakebookNotifications.DataAccess.csproj" />
```
formatting

escalonn commented on FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs:
```diff
@@ -0,0 +1,255 @@
+                .Returns(_mockDB.Object);
+
+            //Act
+            var context = new NotificationsContext(_mockSettings.Object); //create context and test constructor
```
what does the mongoclient have to do with this test?

escalonn commented on FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs:
```diff
@@ -0,0 +1,255 @@
+            var context = new NotificationsContext(_mockSettings.Object); //create context and test constructor
+
+            //Assert
+            Assert.NotNull(context);
```
ctors never return null, so just have an empty 'assert' section with a comment indicating success if no exception

escalonn commented on FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs:
```diff
@@ -0,0 +1,255 @@
+
+            settings = new NotificationsDatabaseSettings()
+            {
+                ConnectionString = "mongodb://localhost:27017",
```
why are we providing a value here that seems to depend on a db running for our unit tests? or if this value isnt being used, then why set it?

escalonn commented on FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs:
```diff
@@ -0,0 +1,255 @@
+
+            //Assert
+            Assert.Equal("123123213213", user.Object.Id);
+        }
```
these tests have several problems.
- any new test in a tdd-fail PR is supposed to fail
- mocking the class you're testing
- arrange/act steps misidentified
- but we don't need to unit test POCO classes anyway.

escalonn commented on FakebookNotifications/FakebookNotifications.WebApi/Startup.cs:
```diff
@@ -27,6 +31,12 @@ public Startup(IConfiguration configuration)
+                Configuration.GetSection(nameof(NotificationsDatabaseSettings)));
+
+            // Create context and repos services
+            services.AddScoped<NotificationsContext>();
```
this code is not relevant to the new tests & doesn't belong in this PR

escalonn commented on FakebookNotifications/FakebookNotifications.WebApi/appsettings.json:
```diff
@@ -1,4 +1,10 @@
+    "NotificationsCollection": "Notifications",
+    "ConnectionString": "mongodb://localhost:27017",
+    "DatabaseName": "NotificationsDb"
+  },
```
also not relevant to this PR

escalonn commented on FakebookNotifications/FakebookNotifications.DataAccess/Models/Notification.cs:
```diff
@@ -0,0 +1,34 @@
+﻿using MongoDB.Bson;
```
if this code was going to be unit tested, it'd need to have `get => throw new NotImplementedException()` for all these so that the tests fail. but classes like this with only auto-properties don't need to be unit tested.
(hence these classes should be added in an unlabelled PR, not this PR, but i'll overlook that)

townerr committed:
Removed unneeded unit tests for POCO classes
Removed unneeded appsettings db config
Fixed MongoContextCreationTest
Removed unneeded MongoClient from tests
Removed db services in startup

townerr committed:
fix formatting
